pkg/controller/predicates/resources/resources.go

```diff
@@ -6,33 +6,26 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/predicate"
 )
 
-var _ predicate.Predicate = DeploymentPredicate{}
-
-type DeploymentPredicate struct {
-	predicate.Funcs
-}
-
-// Update implements default UpdateEvent filter for validating generation change.
-func (DeploymentPredicate) Update(e event.UpdateEvent) bool {
-	if e.ObjectOld == nil || e.ObjectNew == nil {
-		return false
-	}
-
-	oldDeployment, ok := e.ObjectOld.(*appsv1.Deployment)
-	if !ok {
-		return false
+func NewDeploymentPredicate() predicate.Funcs {
```

Comment:
why want to use this instead of the old `func (DeploymentPredicate) Update(e event.UpdateEvent) bool {}`?
thought the old one is more clean, if later to have Create() Delete() implemented on deployment.

Reply:
It can be reverted as not related to the PR, however the `predicate.Funcs` is designed for composition rather than inheritance so I find setting the function more idiomatic and avoid creating a useless struct

Reply:
ok, i leave it to you either keep or remove, 
no objection from me to bring it in this PR.

Reply:
removed